checks/api_endpoints.py

```diff
@@ -0,0 +1,49 @@
+from lib.base import CheckBase, whitelist
+
+
+class CheckDeprecatedAPIEndpoints(CheckBase):
+    enable_parameter = "deprecated-api-endpoints"
+    description = 'check that no deprecated api endpoints are used'
+    default_enabled = True
+
+    def _do_check_(self, m, apis, refs):
+        template = "{} {} is using deprecated endpoint {}.  See {} for details"
+
+        kind = m['kind']
+        name = m['metadata']['name']
+        version = m['apiVersion']
+        ref = refs.get(version, "")
+        err_msg = template.format(kind, name, version, ref)
+
+        assert version not in apis, err_msg
+
+    @whitelist('NetworkPolicy', 'Ingress')
```

Comment:
Did you mean `Route`?

Reply:
Route beta endpoint is not being deprecated with 1.16, or at least it isn't mentioned in the notice.  Is a beta endpoint of Route being deprecated?

Reply:
When you apply an Ingress to OpenShift it is automatically changed to a Route

Reply:
In this instance though we are looking at the manifest as written by upstream projects.  In they have an ingress we care about the api version, but not if they have a route.  If openshift drops support for changing a deprecated ingress api then we would care, but if an upstream has a route defined then we wouldn't care.  Right?

Reply:
So would we actually need both?

Reply:
Only if the route object has api versions that are being deprecated.  The list is here:
https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16

Unless route has already deprecated v1beta1 api version, adding it here would cause a bounce unnecessarily.

If we want to force tenants off beta apis then it could make sense, but that's a bigger set of objects.